src/pages/user/Profile.test.js

```diff
@@ -0,0 +1,153 @@
+
+    expect(component.toJSON()).toMatchSnapshot();
+  });
+
+  it('can get gravatar', () => {
+    const component = renderer.create(<RoutedProfile {...props} />);
+    const { instance } = component.root.findByType(Profile);
+
+    expect(instance).toBeDefined();
+    expect(instance.Gravatar).toEqual(
+      'https://www.gravatar.com/avatar/xxx?s=250'
+    );
+  });
+
+  it('can get website', () => {
+    const component = renderer.create(<RoutedProfile {...props} />);
+    const { instance } = component.root.findByType(Profile);
+
+    expect(instance).toBeDefined();
+    expect(instance.Website).toEqual(
+      <a href="//itsfreezing.com">itsfreezing.com</a>
+    );
+  });
+
+  it('can get locationIcon', () => {
+    const component = renderer.create(<RoutedProfile {...props} />);
+    const { instance } = component.root.findByType(Profile);
+
+    expect(instance).toBeDefined();
+    expect(instance.locationIcon).toEqual(<FontAwesomeIcon icon="city" />);
+  });
+
+  it('can get webIcon', () => {
+    const component = renderer.create(<RoutedProfile {...props} />);
+    const { instance } = component.root.findByType(Profile);
+
+    expect(instance).toBeDefined();
+    expect(instance.webIcon).toEqual(<FontAwesomeIcon icon="globe" />);
+  });
+
+  it('can get biocon', () => {
+    const component = renderer.create(<RoutedProfile {...props} />);
+    const { instance } = component.root.findByType(Profile);
+
+    expect(instance).toBeDefined();
+    expect(instance.bioIcon).toEqual(<FontAwesomeIcon icon="info-circle" />);
+  });
+
+  it('can handleEditor show', () => {
+    const component = renderer.create(<RoutedProfile {...props} />);
+    const { instance } = component.root.findByType(Profile);
+
+    expect(instance).toBeDefined();
+    instance.handleEditor(true);
+    expect(instance.state.editting).toEqual(true);
+  });
+
+  it('can handleEditor hide', () => {
+    const component = renderer.create(<RoutedProfile {...props} />);
+    const { instance } = component.root.findByType(Profile);
+
+    expect(instance).toBeDefined();
+    instance.handleEditor(false);
+    expect(instance.state.editting).toEqual(false);
+  });
+
+  it('can handleSubmit', () => {
+    const component = renderer.create(<RoutedProfile {...props} />);
+    const { instance } = component.root.findByType(Profile);
+
+    expect(instance).toBeDefined();
+    instance.handleSubmit({
+      location: 'WhyNotMinot, CAN',
+      bio: 'FA Cheez Whiz is my favorite.',
+      url: 'itsfreezing.com'
+    });
+    expect(actions.updateProfile).toBeCalledWith({
+      location: 'WhyNotMinot, CAN',
+      bio: 'FA Cheez Whiz is my favorite.',
+      url: 'itsfreezing.com'
+    });
+    expect(instance.state.editting).toEqual(false);
+  });
+
+  it('can display profileEditor', () => {
+    const component = renderer.create(<RoutedProfile {...props} />);
+    const { instance } = component.root.findByType(Profile);
+
+    expect(instance).toBeDefined();
```

Comment:
Maybe `.not.toBeNull()` here? Difference being I believe `null` counts as defined, but it's still a "render nothing" scenario.

Reply:
This is supposed to be `expect(instance.profileEditor)...`. Updated it and applied the `.not.toBeNull` as well. Thanks!